refactor(tables): use pseudo private custom properties

dancormier opened the pull request:
This PR refactors `.s-table` component styling to use pseudo-private custom properties. With this refactor, I didn't flatten the cascade for variants/modifiers _entirely_ as I have with many other refactors. Tables include some very targeted styles (targeting`first`/`last`/`nth` of nested elements) that would've made a  _complete*_ flattening more confusing and hard to maintain than the alternative.

> **Note**
> This PR should result in no visual changes for the `.s-table` component.

---

\* "_complete_" as _consistent_, not as in _total_ or _done_.

dancormier committed:
refactor(tables): use pseudo private custom properties

See https://stackoverflow.atlassian.net/browse/STACKS-66

ondrejkonec commented:
Hi @dancormier 

I see that you are refactoring table component and there is [Issue](https://github.com/StackExchange/Stacks/issues/1125 ) that I tried to fix.

So maybe there is chance now for merging my PR, if you feel it make sense.

dancormier committed:
Merge branch 'develop' into dcormier/table-refactor

dancormier committed:
More PPCP refactoring

dancormier committed:
Fix docs-specific .s-table td styling

dancormier committed:
Remove empty child element declarations

dancormier committed:
Cleanup

giamir commented on lib/css/components/tables.less:
```diff
@@ -1,313 +1,296 @@
+    --_ta-th-p: var(--su8);
+    --_ta-th-ta: left;
+    --_ta-th-va: middle;
+    --_ta-th-w: unset;
```
Looks like a duplicate of line 20

dancormier replied:
Oh, I believe the custom property on line 20 should be `--_ta-td-w`. Good catch!

giamir commented on lib/css/components/tables.less:
```diff
@@ -1,313 +1,296 @@
-            }
-            > *:not(:last-child) {
-                border-right-color: transparent;
+            &:nth-child(2n) {
```
It looks like this selector has higher specificity (`.s-table tbody tr:nth-child(2n)`) now than the `.s-table tr.is-disabled` selector. This prevents the disabled `background-color` from being applied correctly for even rows.

<details>
    <summary>Before</summary>
    
<img width="774" alt="Screenshot 2022-12-13 at 13 38 21" src="https://user-images.githubusercontent.com/6231616/207320283-67a776b5-997c-4724-a673-0e29ed7ca151.png">
</details>

<details>
    <summary>After</summary>
    
<img width="774" alt="Screenshot 2022-12-13 at 13 39 38" src="https://user-images.githubusercontent.com/6231616/207320535-810d82c5-682c-4d6d-b16c-b2b777c53e89.png">
</details>

dancormier replied:
Ah, yep! We need `.is-disabled` to also redefine the custom property setting the even row background color. On it!

giamir commented on docs/assets/less/stacks-documentation.less:
```diff
@@ -318,12 +318,15 @@
+        background-color: var(--black-075);
+        font-family: var(--ff-mono);
+        font-size: var(--fs-caption);
+        color: var(--fc-dark);
```
The specificity of this selector has increased significantly with this change. The result is a slightly different color for the td text in some of the examples. It is negligible and I think there is no action here.
Before the example text was using `color: var(--fc-medium)` (see line 226 in this file) instead of `--fc-dark`.

giamir reviewed (changes requested):
Great work @dancormier!
There are a couple of small fixes to address which I have highlighted in my comments.
Apart from that this looks good to me. I don't think I need to re-review after you have applied the changes - feel free to go ahead and merge. 🙂

dancormier committed:
Fix custom property typo

Addresses https://github.com/StackExchange/Stacks/pull/1204#discussion_r1047073639

dancormier committed:
Fix .is-disabled tr background color override

dancormier committed:
Sort properties alphabetically

dancormier committed:
Renaming tables.less to table.less

dancormier commented:
Thanks for another great review @giamir! It really helps to have your eyes on these. Much like https://github.com/StackExchange/Stacks/pull/1206, I'm holding on merging for now until Core is updated to 1.6.x.

dancormier commented:
Core's been updated. Time to merge!
